src/mixins/window-width-mixin.js

```diff
@@ -0,0 +1,30 @@
+export default {
+  data: () => {
+    return {
+      winWidth: 0,
+    }
+  },
+  mounted() {
+    this.winWidth = window.innerWidth;
+    
+    // timeout reduces the frequency of resize events
+    let respondToResize;
+    window.onresize = () => {
+      clearTimeout(respondToResize);
+      respondToResize = setTimeout(() => {
+        this.winWidth = window.innerWidth;
+      }, 100);
+    };
+  },
+  destroyed() {
+    window.onresize = null;
+  },
+  methods: {
+    windowWidthLessThan(width) {
+      return this.winWidth < width;
+    },
+    windowWidthGreaterThan(width) {
+      return this.winWidth > width;
+    },
```

Comment:
Are these two methods necessary? I would think we could just do the check inline instead of calling an extra method...

Reply:
They aren't necessary, but I felt that accessing a function from outside your component is more industry standard than accessing a variable directly.  Maybe that is just my object oriented design roots coming through.

Reply:
So mixins are a little different. When the component is created and a mixin is provided, the mixin data (props, data, methods, component hooks, etc) are merged in with the component's own properties (options technically). So you're not really calling the methods from "outside" the component. It ends up being the same as calling a method that's attached to the original component. So for a value comparison (`windowWidth < 770` for example), you're comparing values that are already attached to the component context (`this`).